lib/jekyll/static_file.rb

```diff
@@ -19,7 +17,10 @@ def initialize(site, base, dir, name, collection = nil)
       @collection = collection
       @relative_path = File.join(*[@dir, @name].compact)
       @extname = File.extname(@name)
+      # The cache of last modification times [path] -> mtime.
+      @mtimes = {}
```

Comment:
We definitely want this to be a class variable not an instance variable so that all instances share the same Hash. There are alternative ways of implementing a class variable than what we have done. What error prompted this change?